sycl/doc/extensions/Reduction/Reduction.md

```diff
@@ -38,6 +38,30 @@ unspecified reduction(span<T, Extent> var, const T& identity, BinaryOperation co
 
 The exact behavior of a reduction is specific to an implementation; the only interface exposed to the user is the set of functions above, which construct an unspecified `reduction` object encapsulating the reduction variable, an optional operator identity and the reduction operator.  For user-defined binary operations, an implementation should issue a compile-time warning if an identity is not specified and this is known to negatively impact performance (e.g. as a result of the implementation choosing a different reduction algorithm).  For standard binary operations (e.g. `std::plus`) on arithmetic types, the implementation must determine the correct identity automatically in order to avoid performance penalties.
 
+Whether an implementation can identify the identity value for a given combination of accumulator type `AccumulatorT` and function object type `BinaryOperation` can be determined using the `has_known_identity` trait class:
+```c++
+template <typename BinaryOperation, typename AccumulatorT>
+struct has_known_identity {
+  static constexpr bool value;
+};
+
+// Available if C++17
+template <typename BinaryOperation, typename AccumulatorT>
+inline constexpr bool has_known_identity_v = has_known_identity<BinaryOperation, AccumulatorT>::value;
+```
+
+If `has_known_identity` returns `true` for a given combination of accumulator type and function object type, the value of the identity can be extracted using the `known_identity` trait class:
+```c++
+template <typename BinaryOperation, typename AccumulatorT>
+struct known_identity {
+  static constexpr T value;
```

Comment:
What is `T`?

Reply:
Typo. Should be `AccumulatorT`.

Reply:
Oops.  Thanks, both: fixed in c2ff9dcde43c.